_posts/2022/2022-03-14-where-first-vs-find-by-know-what-your-scopes-are-doing.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "Know what your scopes are doing: where.first vs find_by"
+description: "Attack of the abstracted SQL"
+layout: article
+category: ruby
+image:
+  base: "2022/where-first-vs-find-by-know-what-your-scopes-are-doing"
+  alt: "Wooden file card drawers"
+  credit: "Jan Antonin Kolar"
+  source: "https://unsplash.com/photos/lRoX0shwjUQ"
+
+---
+
+There are numerous benefits to using an Object-Relational Mapper (ORM), such as Active Record, to generate your SQL. You gain from clear repeatable abstractions, saving time and gaining readability.
+
+Unfortunately the very abstractions that make an ORM useful can also have unintended consequences.
+
+It is very easy to introduce performance problems because the _actual_ SQL that runs on your database server is generated by the framework.
```

Comment:
It's because the SQL generated is often not the most efficient, right? Just because it's generated by the framework doesn't mean it's inherently bad, right? If so, I think this paragraph/sentence needs a little tweaking to make that clear.

Reply:
```suggestion
It is very easy to introduce performance problems because the _actual_ SQL that runs on your database server is generated by the framework, sometimes this means the SQL isn’t as efficient as it could be.
```